<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>620</width>
    <height>436</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>620</width>
    <height>436</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>620</width>
    <height>436</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>图像几何变换</string>
  </property>
  <widget class="QWidget" name="mainWidget">
   <widget class="QTabWidget" name="argTabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>600</width>
      <height>110</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>600</width>
      <height>110</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>600</width>
      <height>110</height>
     </size>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="translateTab">
     <attribute name="title">
      <string>平移变换</string>
     </attribute>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>86</x>
        <y>10</y>
        <width>391</width>
        <height>64</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="transXTip">
           <property name="text">
            <string>横向平移量（向右为正，向左为负，取整数，单位像素）：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="transXVal"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="transYTip">
           <property name="text">
            <string>纵向平移量（向上为正，向下为负，取整数，单位像素）：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="transYVal"/>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="mirrorTab">
     <attribute name="title">
      <string>镜像变换</string>
     </attribute>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>20</y>
        <width>311</width>
        <height>22</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="mirrorTip">
         <property name="text">
          <string>选择镜像方向：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="mirDirectionVal">
         <item>
          <property name="text">
           <string>垂直镜像（X轴为翻转轴）</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>水平镜像（Y轴为翻转轴）</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="shearTab">
     <attribute name="title">
      <string>错切变换</string>
     </attribute>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>10</y>
        <width>344</width>
        <height>50</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QLabel" name="shearDirectionTip">
           <property name="text">
            <string>错切方向：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="shearTip">
           <property name="text">
            <string>错切量（可取正整数，单位像素）：</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <widget class="QComboBox" name="shearDirectionVal">
           <item>
            <property name="text">
             <string>横向错切（向右）</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>纵向错切（向下）</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="shearVal"/>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="scaleTab">
     <attribute name="title">
      <string>缩放变换</string>
     </attribute>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>84</x>
        <y>10</y>
        <width>391</width>
        <height>64</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_6">
         <item>
          <widget class="QLabel" name="scaleXTip">
           <property name="text">
            <string>X轴缩放量（变长大于1，变短小于1，取正实数，单位倍）：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="scaleXVal"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <widget class="QLabel" name="scaleYTip">
           <property name="text">
            <string>Y轴缩放量（变长大于1，变短小于1，取正实数，单位倍）：</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="scaleYVal"/>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="rotateTab">
     <attribute name="title">
      <string>旋转变换</string>
     </attribute>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>20</y>
        <width>393</width>
        <height>28</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <item>
        <widget class="QLabel" name="rotateTip">
         <property name="text">
          <string>旋转角度（顺时针方向为正，取整数，单位度）：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="rotateVal"/>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
   <widget class="QGroupBox" name="resultBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>170</y>
      <width>600</width>
      <height>239</height>
     </rect>
    </property>
    <property name="title">
     <string>处理结果</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>12</x>
       <y>16</y>
       <width>581</width>
       <height>211</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>进度:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QProgressBar" name="progressBar">
          <property name="value">
           <number>24</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_11">
        <item>
         <widget class="QLabel" name="imgBefLabel">
          <property name="maximumSize">
           <size>
            <width>286</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(85, 170, 127);
</string>
          </property>
          <property name="text">
           <string>变换前图片</string>
          </property>
          <property name="scaledContents">
           <bool>true</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="imgAftLabel">
          <property name="maximumSize">
           <size>
            <width>285</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 127);</string>
          </property>
          <property name="text">
           <string>变换后图片</string>
          </property>
          <property name="scaledContents">
           <bool>true</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="Line" name="hLine1">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>111</y>
      <width>601</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="hLine2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>600</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>130</y>
      <width>591</width>
      <height>28</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_9">
     <item>
      <widget class="QPushButton" name="startBtn">
       <property name="text">
        <string>开始</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <zorder>resultBox</zorder>
   <zorder>hLine1</zorder>
   <zorder>hLine2</zorder>
   <zorder>argTabWidget</zorder>
   <zorder>layoutWidget</zorder>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>620</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="File">
    <property name="title">
     <string>文件(&amp;F)</string>
    </property>
    <addaction name="openAction"/>
    <addaction name="separator"/>
    <addaction name="exitAction"/>
   </widget>
   <widget class="QMenu" name="Operate">
    <property name="title">
     <string>操作</string>
    </property>
    <addaction name="translateAction"/>
    <addaction name="mirrorAction"/>
    <addaction name="shearAction"/>
    <addaction name="scaleAction"/>
    <addaction name="rotateAction"/>
   </widget>
   <widget class="QMenu" name="Help">
    <property name="title">
     <string>帮助</string>
    </property>
    <addaction name="aboutAction"/>
   </widget>
   <addaction name="File"/>
   <addaction name="Operate"/>
   <addaction name="Help"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="openAction">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>打开文件</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="openFolderAction">
   <property name="text">
    <string>打开文件夹</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="aboutAction">
   <property name="text">
    <string>关于</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="saveAction">
   <property name="text">
    <string>保存</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="exitAction">
   <property name="text">
    <string>退出</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="translateAction">
   <property name="text">
    <string>平移变换</string>
   </property>
  </action>
  <action name="scaleAction">
   <property name="text">
    <string>缩放变换</string>
   </property>
  </action>
  <action name="mirrorAction">
   <property name="text">
    <string>镜像变换</string>
   </property>
  </action>
  <action name="shearAction">
   <property name="text">
    <string>错切变换</string>
   </property>
  </action>
  <action name="rotateAction">
   <property name="text">
    <string>旋转变换</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
